edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerCommandFactory.cs

```diff
@@ -45,6 +46,22 @@ public async Task<ICommand> CreateAsync(IModuleWithIdentity module, IRuntimeInfo
             return NullCommand.Instance;
         }
 
+        public Task<ICommand> PrepareUpdateAsync(IModule module, IRuntimeInfo runtimeInfo)
+        {
+            if (module is DockerModule dockerModule)
+            {
+                CombinedDockerConfig combinedDockerConfig = this.combinedConfigProvider.GetCombinedConfig(dockerModule, runtimeInfo);
+
+                if (module.ImagePullPolicy != ImagePullPolicy.Never)
+                {
+                    return Task.FromResult(new PullCommand(this.client, combinedDockerConfig) as ICommand);
+                }
+            }
+
+            return Task.FromResult(NullCommand.Instance as ICommand);
+        }
+
+        // TODO ANDREW: Don't pull on create or update
```

Comment:
This whole file will be removed. First this PR needs to be merged:
https://github.com/Azure/iotedge/pull/6452